core/src/main/java/io/lionweb/lioncore/java/serialization/PrimitiveValuesSerialization.java

```diff
@@ -173,7 +188,74 @@ public String serialize(@Nonnull String primitiveTypeID, @Nullable Object value)
     }
   }
 
+  /** Please note that this will require support for reflection. */
+  public <E extends Enum<E>> void registerEnumClass(Class<E> enumClass, Enumeration enumeration) {
+    primitiveSerializers.put(enumeration.getID(), serializerFor(enumClass, enumeration));
+    primitiveDeserializers.put(enumeration.getID(), deserializerFor(enumClass, enumeration));
+  }
+
   private boolean isEnum(String primitiveTypeID) {
     return enumerationsByID.containsKey(primitiveTypeID);
   }
+
+  public static <E extends Enum<E>> PrimitiveSerializer<E> serializerFor(
+      Class<E> enumClass, Enumeration enumeration) {
+    return value -> {
+      String enumerationLiteralName = value.name();
+      Optional<EnumerationLiteral> enumerationLiteral =
+          enumeration.getLiterals().stream()
+              .filter(l -> l.getName().equals(enumerationLiteralName))
```

Comment:
We identify enum literals by key, not by name.

Reply:
Yes, we use the key for serialization and deserialization (see for example the return statement of this method). Here however we are trying to map an enumerationLiteral to a particular enum constant from an Enum class. In that case we use the name to pair them.

In other words, let's say we have an enum class MyEnum:

```
enum MyEnum { FOO, BAR }
``` 

In this method we register a primitive serializer that associates instances of MyEnum with a certain Enumeration. In this serializer we will receive an instance of MyEnum, for example `FOO`. We will then look for an enumeration literal named `FOO` among the literals of the enumeration associated with the enum class. But then, once we have the enumeration literal, we use its key for producing the serialization value.

Reply:
ok, I get it. Names must always be unique in Java, so it should work.